support-frontend/assets/helpers/productPrice/__tests__/paperProductPricesTests.js

```diff
@@ -0,0 +1,223 @@
+          },
+        },
+      },
+      Sunday: {
+        Monthly: {
+          GBP: {
+            price: 10.79, savingVsRetail: 13, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      WeekendPlus: {
+        Monthly: {
+          GBP: {
+            price: 12.56, savingVsRetail: 26, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      Everyday: {
+        Monthly: {
+          GBP: {
+            price: 47.62, savingVsRetail: 29, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      EverydayPlus: {
+        Monthly: {
+          GBP: {
+            price: 29.20, savingVsRetail: 41, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+    },
+    HomeDelivery: {
+      SixdayPlus: {
+        Monthly: {
+          GBP: {
+            price: 30.36, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      SundayPlus: {
+        Monthly: {
+          GBP: {
+            price: 26.39, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      SaturdayPlus: {
+        Monthly: {
+          GBP: {
+            price: 25.96, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      Saturday: {
+        Monthly: {
+          GBP: {
+            price: 14.69, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      Sixday: {
+        Monthly: {
+          GBP: {
+            price: 54.12, savingVsRetail: 5, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      Weekend: {
+        Monthly: {
+          GBP: {
+            price: 25.09, savingVsRetail: 2, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      Sunday: {
+        Monthly: {
+          GBP: {
+            price: 15.12,
+            currency: 'GBP',
+            fixedTerm: false,
+            promotions: [{
+              name: 'examplePromo',
+              description: 'an example promotion',
+              promoCode: 1234,
+              introductoryPrice: {
+                price: 6.99,
+                periodLength: 3,
+                periodType: 'issue',
+              },
+            }],
+          },
+        },
+      },
+      WeekendPlus: {
+        Monthly: {
+          GBP: {
+            price: 12.57, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      Everyday: {
+        Monthly: {
+          GBP: {
+            price: 62.79, savingVsRetail: 9, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+      EverydayPlus: {
+        Monthly: {
+          GBP: {
+            price: 29.19, currency: 'GBP', fixedTerm: false, promotions: [],
+          },
+        },
+      },
+    },
+  },
+};
```

Comment:
just a general comment as I can see that you have moved this from elsewhere.  If it was a new test, I would suggest using a smaller, more focussed data structure for each test, to make it clearer which values were needed.

Reply:
Yeah this data structure was moved and I had then used the same data structure for subsequent tests in this file, which I can change if you think it would be clearer? Though for the original test that was moved i.e. `getMaxSavingVsRetail` all of the data is being used to do the calculation.